compiler/semtempl.nim

```diff
@@ -558,6 +558,10 @@ proc semTemplateDef(c: PContext, n: PNode): PNode =
     incl(s.flags, sfGlobal)
   else:
     s = semIdentVis(c, skTemplate, n.sons[0], {})
+
+  if s.owner != nil and s.owner.name.s == "system" and s.name.s in ["!=", ">=", ">", "incl", "excl", "in", "notin", "isnot"]:
```

Comment:
Why there's this blacklist? Please add a comment (and please try to break down the line to ~100 chars, not everyone has ultra-wide screens. Maybe the whole list can be lifted to its own `const` variable).

Reply:
It's not a black list, it's a white list, and yes I will add a comment about it. This special treatment should make stack traces have the line information to contain the location of the != symbol, not the location of the == symbol in the template declaration. And this is a whitelist of templates where people don't to get detailed internal line information from.

Reply:
Oh well, I misread `in` as `notin`.

> contain the location of the != symbol, not the location of the == symbol in the template declaration

Hmm, if something goes wrong in `==` I think that showing `!=` is quite misleading.

Reply:
if something goes wring in == it could be the == from the expanded !=. In this case you would want the line information from the != where the template is expanded, not the lineinfo from system.nim that has basically no valuable information.